src/app.ts

```diff
@@ -13,14 +13,38 @@ interface TemperatureSummary {
   average: number
 }
 
+const temperatures: Array<TemperatureReading> = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  temperatures.push(...readings);
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
-}
+  const filteredTemperatures = temperatures.filter(value => value.time.getTime() === new Date(date).getTime() && value.city === city);
+  const totalFilteredTemperatures = filteredTemperatures.length;
+
+  if (totalFilteredTemperatures > 0) {
+    const summary: TemperatureSummary = {
+      first: filteredTemperatures[0].temperature,
+      last: filteredTemperatures[totalFilteredTemperatures - 1].temperature,
+      high: filteredTemperatures[0].temperature,
+      low: filteredTemperatures[0].temperature,
+      average: filteredTemperatures[0].temperature,
+    };
+
+    for (let i = 1; i < totalFilteredTemperatures; i++) {
```

Comment:
For readability prefer the use of `forEach` in these cases. And if you can't use `forEach` (for async/await reasons), use `for of` instead.

Even if you use this kind of for, in most cases, you should start with `i = 0`.